db/version_builder.cc

```diff
@@ -388,11 +392,29 @@ class VersionBuilder::Rep {
       return Status::Corruption("VersionBuilder", oss.str());
     }
 
+    // Note: in C++14, this could be done in a more elegant way using
+    // generalized lambda capture.
+    VersionSet* const vs = version_set_;
+    const ImmutableCFOptions* const ioptions = ioptions_;
+
+    auto deleter = [vs, ioptions](SharedBlobFileMetaData* shared_meta) {
+      if (vs) {
+        assert(ioptions);
+        assert(!ioptions->cf_paths.empty());
+        assert(shared_meta);
+
+        vs->AddObsoleteBlobFile(shared_meta->GetBlobFileNumber(),
```

Comment:
So blob file is always in the `cf_paths.front()`?

Reply:
Yes, that's the plan for now at least, since blob files are not associated with levels. We can always add multiple paths support later if needed.